chef/automate
# Platform And Profile version wildcard filtering

### :clipboard: TODO
- [x] elasticsearch migration
- [x] Create a new filtering field 'platform_with_version'
- [x] Update the UI to use 'platform_with_version'. 
- [x] API Tests for the 'platform_with_version' filter type
- [x] Create a new filtering field 'profile_with_version'
- [x] Update the UI to use field 'profile_with_version'
- [x] Add migration for profile.full to fill the field from current data
- [x] Return the profile.full and platform.full field on node requests
- [x] Test for profile.full and platform.full field on node requests
- [x] migration tests

### :nut_and_bolt: Description: What code changed, and why?
Adding the ability to filter reports based on a platform and profile names and version numbers. 
This changes the UI and the compliance-service. The compliance-service adds two new filter types with the name of `platform_with_version` and `profile_with_version`. These allow suggestions and filtering of reports with the versions combined with the name of the item. 

These changes involved a migration that creates the profile full name (that is the profile title and version combined) and allows suggestions on the platform's full name. This migration works all the back to Automate 1. 

Added the full name to the platform and profile fields for API requests. 

### :chains: Related Resources
#633 
#744

### :+1: Definition of Done
Can filter and get suggestions of nodes from the platform and profile name and version in the compliance reporting search bar. 

### :athletic_shoe: How to Build and Test the Change
1. `build components/compliance-service && build components/automate-gateway && build components/automate-ui && start_all_services`
1. `chef_load_compliance_nodes 10`
1. Go to https://a2-dev.test/compliance/reports/nodes
1. In the search bar select "Platform" and type one character. 
1. Ensure the suggestions have version numbers in them
1. Type all a platform's name and the first number of the version and then a "\*" and press enter. Example "ubuntu 12*"
1. Ensure only the proper nodes are shown. 
1. Do the same for the profile type in the search bar. 

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
